client/src/pages/Reports/Reports.tsx

```diff
@@ -13,10 +14,16 @@ import HeaderWithIcon from 'components/PageTemplate/components/HeaderWithIcon/He
 import LinkComponent from 'components/LinkComponent/LinkComponent.tsx';
 import Spinner from 'components/Spinner/Spinner.tsx';
 import {dateFromTimestampMs} from 'lib/date.ts';
+import './Reports.scss';
+import CustomTag from 'components/CustomTag/CustomTag.tsx';
 
 const Reports = () => {
-  const [rows, setRows] = useState<ReportSummary[]>([]);
-  const [loading, setLoading] = useState<boolean>(true);
+  const [rowsOnDemand, setRowsOnDemand] = useState<ReportSummary[]>([]);
+  const [rowsScheduled, setRowsScheduled] = useState<ReportSummary[]>([]);
+  const [rowsAwaitingGeneration, setRowsAwaitingGeneration] = useState<AwaitingGenerationReport[]>([]);
+  const [loadingOnDemand, setLoadingOnDemand] = useState<boolean>(true);
+  const [loadingScheduled, setLoadingScheduled] = useState<boolean>(true);
+  const [loadingGenerating, setLoadingGenerating] = useState<boolean>(true);
```

Comment:
Consider renaming loadingGenerating arePendingReportsLoading or something similiar. We use this kind of naming in other services

Reply:
There are quite a lot of different states in this component. Consider splitting it into separate components or consider joining data which is being loaded with the isLoading parameter itself (somehting like {data: ..., isLoading:boolean}